charts/argo-workflows/templates/controller/workflow-sa.yaml

```diff
@@ -1,13 +1,18 @@
 {{- if .Values.workflow.serviceAccount.create -}}
+  {{- range $namespace := .Values.singeNamespace | ternary (list "") (append .Values.controller.workflowNamespaces .Values.workflow.namespace | compact | uniq) }}
+---
 apiVersion: v1
 kind: ServiceAccount
 metadata:
-  name: {{ .Values.workflow.serviceAccount.name }}
-  {{- with .Values.workflow.namespace }}
+    {{- with $.Values.workflow.serviceAccount.name }}
+  name: {{ . }}
```

Comment:
There is no need to wrap name in `{{with}}` as `values.yaml` provides a valid default for `.Values.workflow.serviceAccount.name`.

Reply:
Fixed in https://github.com/argoproj/argo-helm/pull/824/commits/531872185a083ea52d990830f4b77afc0b132dcb